File: Data Structures & Algorithms/combinations/submission-1.cpp
class Solution {
private:
    vector<int> path;
    vector<vector<int>> res;
    void backtracking(int n, int k, int startidx){
        if (path.size() == k){
        	res.push_back(path);
            return;
            }
        for (int i = startidx; i <=n; i++){
            path.push_back(i);
            backtracking(n,k,i+1);
            path.pop_back();
        }
    }
public:
    vector<vector<int>> combine(int n, int k) {
        backtracking(n,k,1);
        return res;

    }
};

File: Data Structures & Algorithms/permutations-ii/submission-1.py
class Solution:
    def backtracking(self, nums: List[int], used: List[int], path: List[int], res: List):
        if len(path) == len(nums):
            res.append(path[:])
            return
        for i in range(len(nums)):
            if i > 0 and nums[i] == nums[i-1] and not used[i-1]:
                continue
            if used[i]:
                continue
            used[i] = 1
            path.append(nums[i])
            self.backtracking(nums, used, path, res)
            used[i] = 0
            path.pop()

    def permuteUnique(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        res = []
        self.backtracking(nums, [0]*len(nums), [], res)
        return res
        